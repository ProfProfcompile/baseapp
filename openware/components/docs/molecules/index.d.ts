export * from './InputBlock/InputBlock';
export * from './OrderInput/OrderInput';
export * from './Order/Order';
export * from './OrderForm/OrderForm';
export * from './TabPanel/TabPanel';
export * from './QRCode/QRCode';
export * from './Table/Table';
export * from './WalletItem/WalletItem';
export * from './DepositCrypto/DepositCrypto';
export * from './FilterInput/FilterInput';
export * from './Dropdown/Dropdown';
export * from './History/History';
export * from './OpenOrders/OpenOrders';
export * from './OrderBook/OrderBook';
export * from './Markets/Markets';
export * from './GridItem/GridItem';
export * from './WalletTradeItem/WalletTradeItem';
