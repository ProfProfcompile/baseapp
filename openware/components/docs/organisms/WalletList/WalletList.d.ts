import * as React from 'react';
import { WalletItemProps } from '../../molecules';
interface WalletListProps {
    walletItems: WalletItemProps[];
    /**
     * Callback function which is invoked whenever wallet item is clicked
     */
    onWalletSelectionChange(item: WalletItemProps): void;
}
interface CompositionState {
    activeIndex: number;
}
/**
 * Component to display list of user wallets. It is scrollable and reacts on WalletItem click.
 */
export declare class WalletList extends React.Component<WalletListProps, CompositionState> {
    constructor(props: WalletListProps);
    itemState: (i: number) => boolean;
    makeWalletItem: (props: WalletItemProps, i: number) => JSX.Element;
    handleClick: (i: number, props: WalletItemProps) => void;
    render(): JSX.Element;
}
export { WalletListProps, };
