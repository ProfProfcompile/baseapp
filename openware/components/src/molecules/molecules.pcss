@import "InputBlock/InputBlock.pcss";
@import "OrderInput/OrderInput.pcss";
@import "Order/Order.pcss";
@import "OrderForm/OrderForm.pcss";
@import "TabPanel/TabPanel.pcss";
@import "QRCode/QRCode.pcss";
@import "WalletItem/WalletItem.pcss";
@import "DepositCrypto/DepositCrypto.pcss";
@import "FilterInput/FilterInput.pcss";
@import "Dropdown/Dropdown.pcss";
@import "Table/Table.pcss";
@import "History/History.pcss";
@import "OpenOrders/OpenOrders.pcss";
@import "OrderBook/OrderBook.pcss";
@import "Markets/Markets.pcss";
@import "GridItem/GridItem.pcss";
@import "WalletTradeItem/WalletTradeItem.pcss";
