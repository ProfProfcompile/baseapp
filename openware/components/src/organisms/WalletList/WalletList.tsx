import * as React from 'react';
import { WalletItem, WalletItemProps } from '../../molecules';

interface WalletListProps {
    walletItems: WalletItemProps[];
    /**
     * Callback function which is invoked whenever wallet item is clicked
     */
    onWalletSelectionChange(item: WalletItemProps): void;
}

interface CompositionState {
    activeIndex: number;
}

const removeAlt = (str: string): string => str.replace('-alt', '');

const style: React.CSSProperties = {
    listStyleType: 'none',
    padding: 'calc(var(--gap) * 0.5) calc(var(--gap))',
};

/**
 * Component to display list of user wallets. It is scrollable and reacts on WalletItem click.
 */
export class WalletList
    extends React.Component<WalletListProps, CompositionState> {

    constructor(props: WalletListProps) {
        super(props);
        this.state = { activeIndex: 0 };
    }
    public itemState = (i: number) => {
        return this.state.activeIndex === i;
    };
    public makeWalletItem = (props: WalletItemProps, i: number) => (
        <li
            key={i}
            style={style}
            onClick={this.handleClick.bind(this, i, props)}
        >
            <WalletItem
                key={i}
                {...{
                    ...props,
                    active: this.itemState(i),
                    currency: removeAlt(props.currency),
                }}
            />
        </li>
    );
    public handleClick = (i: number, props: WalletItemProps) => {
        if (this.props.onWalletSelectionChange) {
            this.props.onWalletSelectionChange(props);
        }
        this.setState({ activeIndex: i });
    };

    public render() {
        return (
            <ul className="cr-wallet-list">
                {this.props.walletItems.map(this.makeWalletItem)}
            </ul>
        );
    }
}

export {
    WalletListProps,
};
