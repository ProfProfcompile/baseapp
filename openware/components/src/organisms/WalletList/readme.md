WalletList example:

```js
const fixtures = require("./fixtures");

const onWalletSelectionChange = (wallet) => console.log(wallet);
<WalletList onWalletSelectionChange={onWalletSelectionChange} walletItems={fixtures.walletItems(10)}/>
```