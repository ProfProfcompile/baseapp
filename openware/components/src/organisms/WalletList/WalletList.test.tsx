import { shallow, ShallowWrapper } from 'enzyme';
import * as React from 'react';
import { SinonSpy, spy } from 'sinon';
import { WalletItemProps } from '../../molecules';
import { WalletList, WalletListProps } from './WalletList';

const onWalletSelectionChange = spy();
const walletItems: WalletItemProps[] = [
    {
        active: false,
        address: 'something-0',
        locked: 1,
        fee: 0.123,
        currency: 'BTC',
        balance: 456,
        type: 'fiat',
    }, {
        active: false,
        address: 'something-1',
        fee: 0.123,
        locked: 100,
        currency: 'USD',
        balance: 456,
        type: 'coin',
    }, {
        active: false,
        address: 'something-2',
        fee: 0.3,
        locked: 0.4,
        currency: 'BTC',
        balance: 2,
        type: 'fiat',
    },
];

const defaultProps: WalletListProps = {
    onWalletSelectionChange: onWalletSelectionChange,
    walletItems: walletItems,
};

const setup = (props: Partial<WalletListProps> = {}) =>
    shallow(<WalletList {...{ ...defaultProps, ...props }} />);

describe('WalletList', () => {
    let wrapper: ShallowWrapper;

    beforeEach(() => {
        wrapper = setup();
    });

    it('should render', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should have correct className', () => {
        expect(wrapper.hasClass('cr-wallet-list')).toBeTruthy();
    });

    it('should handle onWalletSelectionChange callback when an element is pressed', () => {
        const first = wrapper.find('[onClick]').first();
        first.simulate('click');
        expect((onWalletSelectionChange as SinonSpy).calledOnceWith(walletItems[0])).toBeTruthy();
    });

    it('should do nothing if onWalletSelectionChange is undefined', () => {
        wrapper = setup({ onWalletSelectionChange: undefined });
        wrapper.find('[onClick]').last().simulate('click');
        expect(wrapper.state()).toEqual({ activeIndex: 2 });
    });
});
