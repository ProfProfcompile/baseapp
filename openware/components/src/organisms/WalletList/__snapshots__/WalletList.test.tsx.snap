// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WalletList should render 1`] = `
<ul
  className="cr-wallet-list"
>
  <li
    key="0"
    onClick={[Function]}
    style={
      Object {
        "listStyleType": "none",
        "padding": "calc(var(--gap) * 0.5) calc(var(--gap))",
      }
    }
  >
    <WalletItem
      active={true}
      address="something-0"
      balance={456}
      currency="BTC"
      fee={0.123}
      key="0"
      locked={1}
      type="fiat"
    />
  </li>
  <li
    key="1"
    onClick={[Function]}
    style={
      Object {
        "listStyleType": "none",
        "padding": "calc(var(--gap) * 0.5) calc(var(--gap))",
      }
    }
  >
    <WalletItem
      active={false}
      address="something-1"
      balance={456}
      currency="USD"
      fee={0.123}
      key="1"
      locked={100}
      type="coin"
    />
  </li>
  <li
    key="2"
    onClick={[Function]}
    style={
      Object {
        "listStyleType": "none",
        "padding": "calc(var(--gap) * 0.5) calc(var(--gap))",
      }
    }
  >
    <WalletItem
      active={false}
      address="something-2"
      balance={2}
      currency="BTC"
      fee={0.3}
      key="2"
      locked={0.4}
      type="fiat"
    />
  </li>
</ul>
`;
